lib/elasticbeats.py

```diff
@@ -44,10 +45,15 @@ def render_without_context(source, target):
 
 
 def principal_unit_cache():
-    principal_name = getenv('JUJU_REMOTE_UNIT')
     cache = kv()
+    principal_name = getenv('JUJU_PRINCIPAL_UNIT')
```

Comment:
Please use the `hookenv.principal_unit` charmhelper:

https://github.com/juju/charm-helpers/blob/master/charmhelpers/core/hookenv.py#L208